pkg/server/config.go

```diff
@@ -0,0 +1,47 @@
+package server
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/viper"
+)
+
+// Environment Vars
+const (
+	EnvEnvironment    = "ENVIRONMENT"
+	EnvConfigLocation = "CONFIG_PATH"
+)
+
+// Config exposes Server configuration options
+type Config struct {
+	BasePath       string
+	MetricsEnabled bool
+	Name           string
+	Port           int
+}
+
+// NewConfig constructs a new *server.Config instance with defaults
+func NewConfig() *Config {
+	viper.SetConfigName(getEnvironmentVariable(EnvEnvironment, "review"))
```

Comment:
Delicious! Nice one on adding viper here I meant to this ages ago and completely forgot! It's a nice library

Reply:
❤️

Reply:
Depending on what kinda configuration we need in services we can just json or yaml files and also load env vars from a K8s configmap. Actually if we're using json/yaml we should be putting them in a configmap too. We can use something like this:

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: service-xyz-config
  labels:
    k8s-app: service-xyz
    kubernetes.io/cluster-service: 'true'
data:
  SERVER_HOST: service-xyz
  SERVER_PORT: 4200
  METRICS_ENABLED: false
```

And then in deployment.yaml container spec add
```
...
        envFrom:
        - configMapRef:
            name: eros-config
 ```

And for the yaml we'd just use a volume mount. I have a bunch of helm shit lying around I should add a chart for this. Need to actually get my paws on the kubeconfig from you

Reply:
Hmmmm, I was thinking of checking in the configs and using private key decryption in the CI so we don't need to worry about any of this?

Reply:
Hmm, I was thinking of encrypting the configs, and checking them in to version control and them decrypting them in the CI using private keys and passing them to the service at run-time/deployment? 

What's the benefit of using .yml files like this for a config map?

Reply:
Actually :arrow_up:, we're gonna have to use `kubectl apply -f something.yml` for all the `istio` stuff, but imo we should try and only use it for the necessary `istio` stuff

But tbh my mind is easily changed and If you want to use `kubectl apply -f` for anything else, you're obvs more than welcome to :heart: